design/one-pager-crossplane-state-metrics.md

```diff
@@ -0,0 +1,640 @@
+# Count of ready XNetworks
+sum(crossplane_xnetwork_ready_status == 1)
+
+# Count of not-ready XNetworks
+sum(crossplane_xnetwork_ready_status == 0)
+
+# Alert: Specific resource is not ready
+crossplane_xnetwork_ready_status{resource_name="prod-vpc"} == 0
+
+# Alert: Any provider is unhealthy
+crossplane_provider_health{status="False"} > 0
+```
+
+#### Multi-Team Attribution
+
+```promql
+# Ready count per team
+sum by (team) (crossplane_xnetwork_ready_status == 1)
+
+# Not-ready resources by team and region
+sum by (team, region) (crossplane_xnetwork_ready_status == 0)
+
+# Team-specific error rate
+rate(crossplane_xnetwork_synced_status_total{team="data-eng", reason="ReconcileError"}[5m])
+```
+
+#### SLO Tracking
+
+```promql
+# SLO: 99% of XNetworks should be ready
+(sum(crossplane_xnetwork_ready_status == 1) /
+ count(crossplane_xnetwork_ready_status)) * 100 > 99
+
+# Ready percentage per region
+sum by (region) (crossplane_xnetwork_ready_status == 1) /
+  count by (region) (crossplane_xnetwork_ready_status) * 100
+
+# Alert: Team's SLO breach (ready < 95% for 10m)
+(sum by (team) (crossplane_xnetwork_ready_status{team="platform"} == 1) /
+ count by (team) (crossplane_xnetwork_ready_status{team="platform"}) * 100) < 95
+```
+
+#### Capacity Planning and Trends
+
+```promql
+# Total managed resources across all composites
+sum(crossplane_xnetwork_resource_count)
+
+# Average resources per composite by region
+avg by (region) (crossplane_xnetwork_resource_count)
+
+# Rate of resources becoming unhealthy
+rate(crossplane_xnetwork_ready_status_total{ready_status="False"}[5m])
+```
+
+#### Wildcard Monitoring Queries
+
+```promql
+# Total ready resources across all platform API kinds
+sum(crossplane_team_apis_ready_count{status="True"})
+
+# Ready count per API kind
+sum by (kind) (crossplane_team_apis_ready_count{status="True"})
+
+# Which API kinds have failures
+crossplane_team_apis_ready_count{status="False"} > 0
+
+# Rate of resources becoming unhealthy per kind
+rate(crossplane_team_apis_ready_count_total{status="False"}[5m])
+
+# Ready percentage by API kind
+sum by (kind) (crossplane_team_apis_ready_count{status="True"}) /
+  count by (kind) (crossplane_team_apis_ready_count) * 100
+```
+
+#### Troubleshooting and Debugging
+
+```promql
+# Find networks with sync errors
+crossplane_xnetwork_synced_status{reason="ReconcileError"} == 0
+
+# Group sync failures by reason
+sum by (reason) (crossplane_xnetwork_synced_status == 0)
+
+# How many times has a resource flapped between ready/not-ready
+sum(crossplane_xnetwork_ready_status_total{resource_name="prod-vpc"})
+
+# Networks that have never been ready (counter = 0)
+crossplane_xnetwork_ready_status_total{ready_status="True"} == 0
+```
+
+#### Cardinality Management
+
+**User-Controlled Cardinality:**
+
+- Users explicitly define which labels to include via `spec.labels[]`
+- No labels are added by default (except `kind` for wildcard monitoring, `namespace` for namespace metrics)
```

Comment:
Feels like `namespace` could lead to unnecessary cardinality for some users. I can see it being included by default for namespaced resources, but it would be nice to make it disableable.

Reply:
any idea how the api will look for disable ?